src/commonMain/kotlin/exchange.dydx.abacus/functional/vault/VaultDepositWithdrawForm.kt

```diff
@@ -0,0 +1,274 @@
+package exchange.dydx.abacus.functional.vault
+
+import kotlinx.serialization.Serializable
+import kotlin.js.JsExport
+
+@JsExport
+@Serializable
+data class FormData(
+    val action: TransactionAction,
+    val amount: Double?,
+    val acknowledgedSlippage: Boolean,
+    val inConfirmationStep: Boolean,
+)
+
+@JsExport
+@Serializable
+enum class TransactionAction {
+    DEPOSIT,
+    WITHDRAW,
+}
+
+@JsExport
+@Serializable
+data class AccountData(
+    val marginUsage: Double?,
+    val freeCollateral: Double?
+)
+
+@JsExport
+@Serializable
+data class SlippageResponse(
+    val shares: Double,
+    val expectedAmount: Double,
+)
+
+@JsExport
+@Serializable
+data class ValidationError(
```

Comment:
Can we reuse `ValidationError` in `exchange.dydx.abacus.output.input` ?  It has the fields to support localization of error messages.

Reply:
Hmmm I kind of like this smaller type for this case. It's nice that the errors are an enum so the frontends can guarantee they are handling all of the possibilities when abacus upgrades. Most of the fields on the other error object are useless for us here too.

Reply:
In general I don't really like abacus knowing anything about localization

Reply:
For mobile clients, their current pattern is to receive fully localized strings to render directly. @ruixhuang @mike-dydx is it possible to handle the enums within client code to map to strings?

Reply:
unrelated to localization - we could also use a `sealed` type (think union type in TS) for this:

```
sealed interface ValidationResult

data class Success(
  val submissionData 
  ...
) : ValidationResult

data class Error(
  val errors: ...
) : ValidationResult
```


on the localization topic, yeah it's ofc doable, just more boilerplate to write and a bit more work to keep in sync if things change later.

Reply:
I don't think it's a great fit here since one of the errors is a warning (the one about slippage) so present even when things are fine.